samples/client/petstore/typescript-inversify/api/pet.service.ts

```diff
@@ -0,0 +1,296 @@
+        }
+
+        // authentication (petstore_auth) required
+        if (this.APIConfiguration.accessToken) {
+            let accessToken = typeof this.APIConfiguration.accessToken === 'function'
+                ? this.APIConfiguration.accessToken()
+                : this.APIConfiguration.accessToken;
+            headers['Authorization'] = 'Bearer ' + accessToken;
+        }
+        headers['Accept'] = 'application/xml';
+        headers['Content-Type'] = 'application/json';
+
+
+        return this.httpClient.post(`${this.basePath}/pet`, body , headers)
+                    .map(httpResponse => <any>(httpResponse.response))
+                    .toPromise();
+    }
+
+
+    /**
+     * Deletes a pet
+     * 
+     * @param petId Pet id to delete
+     * @param apiKey 
+     
+     */
+
+    public deletePet(petId: number, apiKey?: string, headers: Dictionary<string> = {}): Promise<any> {
+        if (!petId){
+            throw new Error('Required parameter petId was null or undefined when calling deletePet.');
+        }
+        if (apiKey) {
+            headers['api_key'] = String(apiKey);
+        }
+
+        // authentication (petstore_auth) required
+        if (this.APIConfiguration.accessToken) {
+            let accessToken = typeof this.APIConfiguration.accessToken === 'function'
+                ? this.APIConfiguration.accessToken()
+                : this.APIConfiguration.accessToken;
+            headers['Authorization'] = 'Bearer ' + accessToken;
+        }
+        headers['Accept'] = 'application/xml';
+
+
+        return this.httpClient.delete(`${this.basePath}/pet/${encodeURIComponent(String(petId))}`, headers)
+                    .map(httpResponse => <any>(httpResponse.response))
+                    .toPromise();
+    }
+
+
+    /**
+     * Finds Pets by status
+     * Multiple status values can be provided with comma separated strings
+     * @param status Status values that need to be considered for filter
+     
+     */
+
+    public findPetsByStatus(status: Array<'available' | 'pending' | 'sold'>, headers: Dictionary<string> = {}): Promise<Array<Pet>> {
+        if (!status){
+            throw new Error('Required parameter status was null or undefined when calling findPetsByStatus.');
+        }
+        let queryParameters: string[] = [];
+        if (status) {
+            queryParameters.push("status="+status.join(COLLECTION_FORMATS['csv']));
+        }
+
+
+        // authentication (petstore_auth) required
+        if (this.APIConfiguration.accessToken) {
+            let accessToken = typeof this.APIConfiguration.accessToken === 'function'
+                ? this.APIConfiguration.accessToken()
+                : this.APIConfiguration.accessToken;
+            headers['Authorization'] = 'Bearer ' + accessToken;
+        }
+        headers['Accept'] = 'application/xml';
+
+
+        return this.httpClient.get(`${this.basePath}/pet/findByStatus?${encodeURIComponent(queryParameters.join('&'))}`, headers)
```

Comment:
it looks like you need to re-generate the file, since the `encodeURIComponent` should be removed here